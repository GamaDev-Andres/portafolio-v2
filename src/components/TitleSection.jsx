/* eslint-disable multiline-ternary */
import PropTypes from 'prop-types'

// eslint-disable-next-line react/prop-types
const TitleSection = ({ textInSpan, unOrder = false, children }) => {
  return (
    <h2 className="my-4 uppercase font-bold text-3xl md:text-center">
      {unOrder ? (
        <>
          {children} <span className="text-blue-400">{textInSpan}</span>
        </>
      ) : (
        <>
          <span className="text-blue-400">{textInSpan}</span> {children}
        </>
      )}
    </h2>
  )
}

TitleSection.propTypes = {
  textInSpan: PropTypes.string.isRequired,
  unOrder: PropTypes.bool
}
export default TitleSection
