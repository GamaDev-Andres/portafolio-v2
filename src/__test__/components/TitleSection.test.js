/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import TitleSection from '../../components/TitleSection'

describe('probando <TitleSection/>', () => {
  test('debe montarse correctamente', () => {
    const texto = 'prueba'
    const wrapper = Enzyme.mount(
      <TitleSection textInSpan={texto}>
        Primer</TitleSection>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('span').text().trim()).toBe(texto)
  })
})
