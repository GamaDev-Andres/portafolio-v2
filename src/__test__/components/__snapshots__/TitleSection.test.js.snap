// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <TitleSection/> debe montarse correctamente 1`] = `
<h2
  className="my-4 uppercase font-bold text-3xl md:text-center"
>
  <span
    className="text-blue-400"
  >
    prueba
  </span>
   
  Primer
</h2>
`;
